pynestml/cocos/co_co_convolve_cond_correctly_built.py

```diff
@@ -47,13 +47,12 @@ def check_co_co(cls, node):
 
 class ConvolveCheckerVisitor(ASTVisitor):
     """
-    Visits a function call and checks that if the function call is a cond_sum,cur_sum or convolve, the parameters
-    are correct.
+    Visits a function call and checks that if the function call is a convolve, the parameters are correct.
     """
 
     def visit_function_call(self, node):
         func_name = node.get_name()
-        if func_name == 'convolve' or func_name == 'cond_sum' or func_name == 'curr_sum':
+        if func_name == 'convolve':
```

Comment:
I guess I would add a test here to check that if ``symbol_var.is_shape()`` then it is ``real``

Reply:
Right now, in an expression ``convolve(g, spikes_in)``, where ``g`` is the shape and ``spikes_in`` is the spike input port, the type of the expression is equal to the type of the second argument. The type is thus "carried" by the input port, for example if we have a definition ``spikes_in nS <- spike`` then any convolution with ``spikes_in`` will have type nS. The type of the first argument is ignored.

This is suboptimal, and we could change this around to use the type of the first argument (the shape variable) instead, while leaving spike input ports untyped (you would just write ``spikes_in <- spike``). The type of the convolution is then that of the shape state variable, e.g. nS if we define our shape as follows:

```
g nS = 0 nS
g' nS/ms = g_E_const * (1 / tau_syn_rise_E - 1 / tau_syn_decay_E) * nS

shape g'' = -(g + g' * (tau_syn_decay_E + tau_syn_rise_E)) / (tau_syn_decay_E * tau_syn_rise_E)
```